src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -390,7 +391,29 @@ internal void IgnoreCommand(string commandText, InvocationInfo invocation)
         /// so that when content is sent through Out-Default it doesn't
         /// make it to the actual host.
         /// </summary>
-        internal bool TranscribeOnly { get; set; }
+        internal bool TranscribeOnly => Interlocked.CompareExchange(ref _transcribeOnlyCount, 0, 0) != 0;
```

Comment:
I way worried about stale read, so I wanted to read fresh value of variable. But given that stale read can only occur if destructor kick in other thread, which is already nondeterministic, it should be fine with plain read or volatile one.